ocaml/typing/printtyp.ml

```diff
@@ -1867,8 +1867,8 @@ let tree_of_type_decl id decl =
     | Type_variant (cstrs, rep) ->
         let unboxed =
           match rep with
-          | Variant_unboxed -> true
-          | Variant_boxed _ | Variant_extensible -> false
+          | Variant_unboxed  | Variant_with_null -> true
```

Comment:
Won't this cause `[@@unboxed]` to appear in the output? If not, I'm not sure what this is doing.

Reply:
Initially, I thought that `Variant_with_null` is really `Variant_unboxed` and should be handled the same. But, since this isn't true for now, and we have `or_null_reexport`, I decided to add code for printing `or_null_reexport` here.